chronos/asyncfutures2.nim

```diff
@@ -24,14 +30,14 @@ type
   StackTrace = string
 
   FutureBase* = ref object of RootObj ## Untyped future.
+    location: array[2, ptr SrcLoc]
     callbacks: Deque[AsyncCallback]
-
     finished: bool
     error*: ref Exception ## Stored exception
     errorStackTrace*: StackTrace
     stackTrace: StackTrace ## For debugging purposes only.
     id: int
-    fromProc: string
+    fromProc: cstring
```

Comment:
If the proc name is also helpful, you can embed it in the `SourceLocation` struct. The only required change would be to introduce an extra `static string` parameter to `getSrcLocation`. 

The need for such an extra parameter may disappear in the future - It has been suggested few times in the Nim issue tracker that `instantiationInfo` itself can include this information.